### DISCUSSION TITLE
Crash while loading a large directory of FLAC files

### DISCUSSION THREAD

**Discussion by Saijin-Naib** [enzo1982/freac#439] [Help & Support]
I just used a script to dump my BandCamp purchases to FLAC so I can get them encoded to OPUS with FRE:AC to replace my existing MP3 library.

The drive contains 9230 files with 211.8GB size.

I've attached a zip of the gdb log (I do not have a debug release on my distro, unfortunately).
[freac_crash.log.zip](https://github.com/enzo1982/freac/files/10712669/freac_crash.log.zip)

Is there something wrong with the files? Bad Unicode names? Too many? Too large?

**Comment by enzo1982** [comment]
Thank you for reporting this!

Unfortunately, the log does not contain a crash address or stack trace, so it does not allow identifying the cause of the crash.

If you would like to help pin down and fix this issue, here's how you can help:

- If you have been using fre:ac 1.1.6, please try the latest [continuous build](https://github.com/enzo1982/freac/releases/continuous) to check if the crash still happens
- Try to figure out if it crashes on a specific file (I realize this may be difficult with 9230 files, but using a binary search strategy it might be doable)
- If you can identify a specific file, please make it available to me for analysis (you can send a link to support@freac.org)
- If you are versed in using development tools, you can compile fre:ac (and ideally the [smooth](https://github.com/enzo1982/smooth) and [BoCA](https://github.com/enzo1982/BoCA) dependencies too) with debug symbols and run it in GDB to identify the address and source line where the crash happens
- If needed, I can provide a version with debug symbols for you and give you instructions for how to run it in GDB

Generally, fre:ac should be able to handle that many files and the file size should be irrelevant for this. Special characters in a file name should not cause issues either, but I cannot rule that one out completely.

**Reply by Saijin-Naib** [reply]
Thanks for the above!

Might it be failing on the lost+found directory at the root of my WD-Red drive like Sound Converter was doing?:  
https://github.com/kassoulet/soundconverter/pull/68

I would love to use the above builds, but I'm using Alpine Linux latest-stable x86_64, which is MUSL instead of GLIBC, so unfortunately pre-compiled binaries and AppImage will not work.

I might be able to modify the Alpine APKBUILD recipe to enable debug mode. If I go this route, should I target any specific commit or branch?

**Reply by enzo1982** [reply]
The lost+found directory might contain damaged files which could make fre:ac crash. Just trying to add files with restricted access should not cause any issues (although I need to test that to be sure...).

If fre:ac crashes when you try to add the lost+found folder, please try to find out if there is a specific file in that folder causing the crash and make that available to me if possible.

If you could modify the Alpine packaging script to create a debug version, that would be great! You should just use the latest commit on the master branch in that case (same for the smooth and BoCA libraries).

Unfortunately I cannot help you much with getting this to work on Alpine right now, as I currently don't have an Alpine system set up. I may have time to set up a VM in a few days though.

**Comment by Lee-Carre** [comment]
> I just used a script to dump my BandCamp purchases to FLAC

While not ignoring the case of a directory with many files causing software misbehaviour, a more sophisticated approach would be to use [ytdl-org/youtube-dl](/ytdl-org/youtube-dl) (it can fetch from many more sites than YouTube, including BandCamp).

YouTube-DL would (properly configured) extract the metadata to tag the local files with, and can separate groups of files however you wish (artist and/or album, in this case).

YouTube-DL builds are currently being reworked (hence the latest release is quite old, even if the source in the master branch is current). In the meantime, you'll probably want the [nightly builds](/ytdl-org/ytdl-nightly/releases). Unless you're comfortable building from source, or have Python installed in order to run YouTube-DL without compilation/building using your local interpreter.

Since YouTube-DL uses FFmpeg, you may be able to have *it* perform the transcoding, instead of needing Fre:AC.
